[FIX] URI "definition" and recommendation (#629)
bids-standard/bids-specification: src/02-common-principles.md

```JSON
{

>   "SourceDatasets": [ {"URI": "file://../../rawdata/"} ]

Author: Well... That's why we have BIDS 2.0 for ! :rocket:

| Description  | RECOMMENDED           | [string][]                | Description of the column.                                                                                      |
| Levels       | RECOMMENDED           | [object][] of [strings][] | For categorical variables: An object of possible values (keys) and their descriptions (values).                 |
| Units        | RECOMMENDED           | [string][]                | Measurement units. SI units in CMIXF formatting are RECOMMENDED (see [Units](./02-common-principles.md#units)). |

> | TermURL      | RECOMMENDED           | [string][]                | [URI](../02-common-principles.md#uniform-resource indicator) pointing to a formal definition of this type of data in an ontology available on the web.                   |

Author: To me (as a user) it feels awkward o have the keyname mention "URL" but definition asking for an URI.

Reviewer: True, these fields should stay at URL, which is a specific subclass of URI. In the validator, we can still start to validate them as URI --> and it will not break datasets that have been previously valid.

Author: okidoki will revert any of the changes where the keyname contains URL.

}
```

## Uniform Resource Indicator 

> Uniform Resource Indicator (URI) are a string referring to a resource and SHOULD

Reviewer: ```suggestion
A Uniform Resource Indicator (URI) is a string referring to a resource and SHOULD
```

have the form `<scheme>:[//<authority>]<path>[?<query>][#<fragment>]`, as specified 
in [RFC 3986](https://tools.ietf.org/html/rfc3986). 

> This applies tp URLs and other common URIs include Digital Object Identifiers (DOIs), 

Reviewer: ```suggestion
This applies to URLs and other common URIs, including Digital Object Identifiers (DOIs), 
```

}
```

## Uniform Resource Indicator 

A Uniform Resource Indicator (URI) is a string referring to a resource and SHOULD
have the form `<scheme>:[//<authority>]<path>[?<query>][#<fragment>]`, as specified 
in [RFC 3986](https://tools.ietf.org/html/rfc3986). 
This applies to URLs and other common URIs, including Digital Object Identifiers (DOIs), 
which may be fully specified as `doi:<path>`, 
for example, [doi:10.5281/zenodo.3686061](https://doi.org/10.5281/zenodo.3686061).
A given resource may have multiple URIs.
When selecting URIs to add to dataset metadata, it is important to consider 
specificity and persistence.

> 

Author: Does it have to cover URLs specifying the DOI resolver? 

Still trying to wrap my around this: I thought that URLs would be a valid URI anyway.

> 

Author: Yeah I was really not sure about its proper place. Definitely makes sense to have it lower down this hierarchy.

> 

Author: > I'm okay with promoting this above `Units`, but I think `Participant names and other labels` are more fundamental to understanding BIDS concepts and so would push this down below that.

Done

> 

Reviewer 2: `doi:<thing>` and `https://doi.org/<thing>` are equivalent from an identifier perspective. this may be more of a preference as saying  `doi:<thing>` is preferred. bids may decide to not allow the full url formulation. but that seems a bit more draconian.

> 

Reviewer 3: The question is "What do we mean when we say DOI". I would argue that we mean "doi:<DOI>", not an arbitrary URI that may or may not refer to the same object. Currently all the examples show `<DOI>`, but it seemed like a good idea to include `https://doi.org/<DOI>` as a thing that will be warned (which is all DEPRECATED means until BIDS 2.0) about by the validator.

If that feels like it's overstepping, then I'm okay with changing to: 

```Markdown
Bare DOIs such as `10.18112/openneuro.ds000001.v1.0.0` are DEPRECATED.
```

> 

Author: I would say let's make go "small steps" and just have :

```Bare DOIs such as `10.18112/openneuro.ds000001.v1.0.0` are DEPRECATED.```

subject) with the label `01`. The `sub-` prefix is not part of the subject
label, but must be included in file names (similarly to other key names).

## Uniform Resource Indicator 

A Uniform Resource Indicator (URI) is a string referring to a resource and SHOULD
have the form `<scheme>:[//<authority>]<path>[?<query>][#<fragment>]`, as specified 
in [RFC 3986](https://tools.ietf.org/html/rfc3986). 
This applies to URLs and other common URIs, including Digital Object Identifiers (DOIs), 
which may be fully specified as `doi:<path>`, 
for example, [doi:10.5281/zenodo.3686061](https://doi.org/10.5281/zenodo.3686061).
A given resource may have multiple URIs.
When selecting URIs to add to dataset metadata, it is important to consider 
specificity and persistence.

> 

Reviewer 3: ```suggestion

Several fields are designated for DOIs, for example, `DatasetDOI` in `dataset_description.json`.
DOI values SHOULD be fully specified URIs such as `doi:10.18112/openneuro.ds000001.v1.0.0`.
Bare DOIs such as `10.18112/openneuro.ds000001.v1.0.0` are [DEPRECATED].
```

Author: I am completely OK with this formulation and only excluding bare ROI but allowing valid URI in general.
